docs/comlink/reference/map.mdx

```diff
@@ -346,6 +346,161 @@ http POST "/users" {
+
+#### Return & Fail
+Once the operation is done, it can `return` the result value.
+
+Operation can also `return` early or `fail` fast if a condition is met.
+
+```hcl title="Return & Fail"
+operation FulfilOrder {
+  fail if (!args.order.isPaid) {
+    reason = "Cannot fulfill order that's unpaid"
+  }
+  
+  return if (args.order.isFulfilled) {
+    fulfilledOrder = args.order
+  }
+  
+  // ...
+  
+  return {
+    fulfilledOrder = { ... }
+  }
+}
+```
+
+This operation expects `order` keyword argument. It will fail immediately if the order is still in unpaid state. Then, if the order was already fulfilled, it will bypass the subsequent code and early-returns the order itself, making the operation idempotent.
+
+### Calling operation
+```hcl
+call OperationName(argument1=value, argument2=value, ...)
+```
+
+#### Calling in-place
+The most common use of operation is calling it in-place. The operation is called, and in case of success the operation result value is returned by the call.
+
+Operation can be called in-place only when the returned value is assigned a variable (the operation call is on the right-hand side).
+
+```hcl title="In-place operation call" {5}
+map MakeFulfillment {
+  // ...
+  orderData = { ... }
+  
+  fulfilledOrder = call FulfilOrder(order = orderData)
+  
+  // ...
+}
+```
+
+The code above calls `FulfilOrder` operation and stores the result in `fulfilledOrder` variable. The `orderData` is passed as the value for `order` keyword argument.
+
+There is no try-catch mechanism when calling the operation in-place. If the operation runs into `fail` branch, the call fails silently and an empty value (e.g. `undefined`) is assigned as the result of the call.
+
+If the operation has `fail` branch and you need to handle the failure gracefully, use the operation [call with handler](#calling-with-handler).
+
+#### Calling with handler
+Operation can also be called with outcome handler. The operation is called, and the the call outcome is handled inside an explicit code block.
+
+This is useful if the called operation has `fail` branch which needs to be handled gracefully.
+
+The outcome can be accessed via `outcome` context variable inside the handler block:
+- `outcome.data` — the result value returned by `return`
+- `outcome.error` — the error value returned by `fail`
+
+```hcl title="Operation call with handler" {5-14}
+map MakeFulfillment {
+  // ...
+  orderData = { ... }
+  
+  call FulfilOrder(order = orderData) {
+    map error if (outcome.error) {
+      title = "Fulfilment Failed"
+      detail = outcome.error.reason
+    }
+
+    fulfilledOrder = outcome.data.fulfilledOrder
+		
+    // ...
+  }
+}
+```
+The code above calls `FulfilOrder` operation. If the operation errors with `fail`, the `MakeFulfillment` use case is mapped to an expected error including some details. Otherwise, the returned value is assigned to `fulfilledOrder` variable.
```

Comment:
```suggestion
The code above calls `FulfilOrder` operation. If the operation errors out with `fail`, the `MakeFulfillment` use case is mapped to an expected error including its details. Otherwise, the returned value is assigned to the `fulfilledOrder` variable.
```